bpf/kmesh/workload/sendmsg.c

```diff
@@ -185,6 +256,29 @@ int sendmsg_prog(struct sk_msg_md *msg)
 
     // encode org dst addr
     encode_metadata_org_dst_addr(msg, &off, (msg->family == AF_INET));
+
+    struct bpf_sock *sk = msg->sk;
+
+    if (sk) {
+        if (is_managed_by_kmesh_skmsg(msg)) {
+            observe_on_data(sk);
+        }
+    } else {
+        BPF_LOG(ERR, KMESH, "sk_lookup success\n");
+    }
+    int key = 0;
+    __u64 *last_time = bpf_map_lookup_elem(&tcp_conn_last_flush, &key);
+    __u64 now = bpf_ktime_get_ns();
+
+    if (!last_time) {
+        __u64 init_time = now;
+        // Initialize last flush time if not set
+        bpf_map_update_elem(&tcp_conn_last_flush, &key, &init_time, BPF_ANY);
+    } else if ((now - *last_time) >= TIMER_INTERVAL_NS) {
+        flush_tcp_conns();
```

Comment:
IIRC, this is flushing all the connections probe info, i think it is a little bit heavy.

Can you do that for this particular socket?

Reply:
You may get the   bpf_tcp_sock  from   struct sk_msg_md

    struct bpf_sock *sk = msg->sk;
    struct bpf_tcp_sock *tcp_sk = bpf_skc_to_tcp_sock(sk);